docs/CNDP_README.md

```diff
@@ -0,0 +1,199 @@
+
+![Test setup] (./images/cndp-omec-upf-test-setup.jpg)
+
+There are two systems: System 1 is runs CNDP BESS UPF and System 2 runs DPDK based packet generator which simulates traffic generated from multiple UEs and App servers. 
+
+System 1 and System 2 are connected using  physical links. Setup uses two network ports which represents access and core interface. This test setup uses Intel Ethernet 800 series network adapter (hereafter referred to as NIC).  NIC in System 1 has Intel Dynamics Device Personalization (DDP) for telecommunications workload enabled. DDP profile can help in GTPU packet traffic steering to required NIC hardware queues using RSS (Receive Side Scaling). DDP feature works along with XDP offload feature in NIC hardware to redirect GTPU packets directly to user space via AF-XDP sockets. Refer the Deployment section in this [document](https://builders.intel.com/docs/networkbuilders/intel-ethernet-controller-800-series-device-personalization-ddp-for-telecommunications-workloads-technology-guide.pdf) to enable DDP.
+
+Install NIC driver in System 1 from this link :  https://sourceforge.net/projects/e1000/files/ice%20stable/1.9.7/ .  This driver is used instead of in-tree kernel driver since this driver supports tc filter for creating queue groups and RSS for GTPU packet traffic steering. 
+
+The setup uses Physical Function (PF) of PCIe network adaptor (where the driver supports AF-XDP zero copy ) for improved network I/O performance. AF-XDP zero copy support for SR-IOV driver and sub function support using devlink will be supported in future releases.
+
+### Step 3: Run UPF in System 1
+
+1) Setup hugepages in the system 1. For example, to setup 8GB of total huge pages (8 pages each of size 1GB in each NUMA node) using DPDK script, use the command `dpdk-hugepages.py -p 1G --setup 8G`
+
+2) Enable cndp mode in configuration file  [upf.json](./conf/upf.json)
+
+```
+diff --git a/conf/upf.json b/conf/upf.json
+index 5196e94..beec3fd 100644
+--- a/conf/upf.json
++++ b/conf/upf.json
+@@ -5,7 +5,7 @@
+     "": "mode: sim",
+     "": "mode: dpdk",
+     "": "mode: cndp",
+-    "mode": "dpdk",
++    "mode": "cndp",
+```
+
+3) Enable cndp mode in script file  [docker_setup.sh](./scripts/docker_setup.sh)
+
+```
+diff --git a/scripts/docker_setup.sh b/scripts/docker_setup.sh
+index 7aff6a6..1a8e2fd 100755
+--- a/scripts/docker_setup.sh
++++ b/scripts/docker_setup.sh
+@@ -15,8 +15,8 @@ metrics_port=8080
+ # "af_packet" uses AF_PACKET sockets via DPDK's vdev for pkt I/O.
+ # "sim" uses Source() modules to simulate traffic generation
+ # "cndp" use kernel AF-XDP. It supports ZC and XDP offload if driver and NIC supports it. It's tested on Intel 800 series n/w adapter.
+-mode="dpdk"
+-#mode="cndp"
++#mode="dpdk"
++mode="cndp"
+ #mode="af_xdp"
+ #mode="af_packet"
+ #mode="sim"
+```
+
+3) Modify [cndp_upf_1worker.jsonc](./conf/cndp_upf_1worker.jsonc) file lports section to update the access and core netdev interface name and required queue id.
```

Comment:
This should be `4)`, the previous item is already `3)`

Reply:
ok